File: LabOOP2.2/LabOOP2.2/Figure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LabOOP2._2
{
    class Figure
    {
        public readonly List<Point> allPoints;

        public Figure(Point A, Point B, Point C)
        {
            allPoints = new List<Point>();
            allPoints.Add(A);
            allPoints.Add(B);
            allPoints.Add(C);
        }

        public Figure(Point A, Point B, Point C, Point D)
        {
            allPoints = new List<Point>();
            allPoints.Add(A);
            allPoints.Add(B);
            allPoints.Add(C);
            allPoints.Add(D);
        }

        public Figure(Point A, Point B, Point C, Point D, Point E)
        {
            allPoints = new List<Point>();
            allPoints.Add(A);
            allPoints.Add(B);
            allPoints.Add(C);
            allPoints.Add(D);
            allPoints.Add(E);
        }

        protected double Length(Point A, Point B)
        {
            return Math.Sqrt(Math.Pow((A.x - B.x), 2) + Math.Pow((A.y - B.y), 2)); ;
        }
    }
}

File: LabOOP2.2/LabOOP2.2(c++)/Figure.cpp
#include "stdafx.h"
#include "Figure.h"

#include <math.h>

Figure::Figure()
{
}


Figure::~Figure()
{
}

Figure::Figure(Point A, Point B, Point C)
{
	allPoints.push_back(A);
	allPoints.push_back(B);
	allPoints.push_back(C);
}

Figure::Figure(Point A, Point B, Point C, Point D)
{
	allPoints.push_back(A);
	allPoints.push_back(B);
	allPoints.push_back(C);
	allPoints.push_back(D);
}

Figure::Figure(Point A, Point B, Point C, Point D, Point E)
{
	allPoints.push_back(A);
	allPoints.push_back(B);
	allPoints.push_back(C);
	allPoints.push_back(D);
}

double Figure::Length(Point A, Point B)
{
	return sqrt(pow((A.x - B.x), 2) + pow((A.y - B.y), 2));
}
